rpc/types.go

```diff
@@ -0,0 +1,116 @@
+package rpc
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+// https://github.com/starkware-libs/starknet-specs/blob/a789ccc3432c57777beceaa53a34a7ae2f25fda0/api/starknet_api_openrpc.json#L520-L534
+type BlockNumberAndHash struct {
+	Number uint64     `json:"block_number"`
+	Hash   *felt.Felt `json:"block_hash"`
+}
+
+// https://github.com/starkware-libs/starknet-specs/blob/a789ccc3432c57777beceaa53a34a7ae2f25fda0/api/starknet_api_openrpc.json#L814
+type BlockId struct {
+	Pending bool
+	Latest  bool
+	Hash    *felt.Felt
+	Number  uint64
+}
```

Comment:
Was thinking we should add `json` tag here as well, but not sure if it's intentional

```

type BlockId struct {
	Pending bool       `json:"pending,omitempty"`
	Latest  bool       `json:"latest,omitempty"`
	Hash    *felt.Felt `json:"hash,omitempty"`
	Number  uint64     `json:"number,omitempty"`
}
```

But the tests seem to be working fine after adding it

Reply:
Yeap, no tags was intentional. BlockId has custom unmarshaler. Check out how blockid is defined in the spec from the link above it.

Reply:
Right. Got it.